src/include/libminiasync/data_mover_threads.h

```diff
@@ -10,14 +10,16 @@
 extern "C" {
 #endif
 
-struct data_mover_threads;
+typedef void *(*memcpy_fn)(void *dst, const void *src, size_t n);
```

Comment:
make it a struct.

struct data_mover_threads_ops {
  memcpy_fn;
};
and pass it by pointer to the function + a size argument. For example:

data_mover_threads_new(..., &ops, sizeof(ops));

This way we can extend the structure without breaking ABI compatibility.

Also, we don't really need the no-flags memcpy. You can just wrap a libc memcpy function that ignores flags (or does something with them).